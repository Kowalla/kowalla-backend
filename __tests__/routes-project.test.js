const request = require("supertest");
const { mongooseConnect } = require("../helpers/mongoose");
const { appInit } = require("../helpers/express");

describe("routes/project", () => {
  let app;

  beforeAll(async () => {
    // TODO: Kevin 2019-10 Move into global Jest setup
    await mongooseConnect();
    app = appInit();
    return app;
  });

  describe("getProjectList()", () => {
    it("success", async () => {
      const res = await request(app)
        .get("/api/v1/projects")
        .set("Accept", "application/json")
        .expect(200);

      expect(res.body).toMatchSnapshot();
    });
  });
});
