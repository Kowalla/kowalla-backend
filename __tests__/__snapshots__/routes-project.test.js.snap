// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/project getProjectList() success 1`] = `
Object {
  "projects": Array [
    Object {
      "__v": 0,
      "_id": "fS50wWyBY",
      "admins": Array [
        "qkJmx0Ma_P",
      ],
      "createdAt": "2019-09-24T03:25:34.420Z",
      "description": "The world's online coworking space, for testing!",
      "headerPicture": "",
      "id": "fS50wWyBY",
      "isProject": true,
      "name": "kowalla;",
      "postCount": 0,
      "profilePicture": "https://ui-avatars.com/api/?name=Kowalla&background=0a2049&color=fff&bold=true&size=200&font-size=0.6",
      "projectName": "Kowalla",
      "reputation": 0,
      "subscribers": 1,
      "updatedAt": "2019-09-24T03:25:55.480Z",
    },
  ],
}
`;
